packages/react-generator/src/expressions/react-component.ts

```diff
@@ -182,6 +182,25 @@ export class ReactComponent extends Component {
 
   createRestPropsGetter(members: BaseClassMember[]): GetAccessor {
     const props = getProps(members);
+    const additionalBindings = [
+      new BindingElement(
+        SyntaxKind.DotDotDotToken,
+        undefined,
+        new Identifier('restProps'),
+      ),
+    ];
+
+    // TODO: remove it after overlay components are renovated
+    if (isComponentWrapper(this.context.imports)) {
+      additionalBindings.unshift(
+        new BindingElement(
+          undefined,
+          undefined,
+          new Identifier('isReactComponentWrapper'),
+        ),
+      );
+    }
+
     const bindingElements = props
       .reduce((bindingElements, p) => {
```

Comment:
Please change cast: [] as BindingElement[] to more convinent generic: .reduce<BindingElement[]>(